common/containers/Tabs/Contracts/components/Interact/components/InteractExplorer/index.tsx

```diff
@@ -73,19 +74,91 @@ class InteractExplorerClass extends Component<Props, State> {
     outputs: {}
   };
 
+  public setParams = () => {
+    const params = this.getParamsFromUrl();
+    let count = 0;
+    Object.keys(params).map((index: any) => {
+      const item = params[index];
+      if (index.substr(0, 5) === 'param') {
+        count++;
+        const element: any = document.getElementById(index);
+        if (element) {
+          element.value = item;
+          const rawValue: string = item;
+          const isArr = rawValue.startsWith('[') && rawValue.endsWith(']');
+
+          const value = {
+            rawData: rawValue,
+            parsedData: isArr ? this.tryParseJSON(rawValue) : rawValue
+          };
+          this.setState({
+            inputs: {
+              ...this.state.inputs,
+              [element.name]: value
+            }
+          });
+        }
+      }
+    });
+    if (count > 0) {
+      const itemToScrollTo = document.getElementById('contractRead');
+      if (itemToScrollTo) {
+        itemToScrollTo.scrollIntoView();
+      }
+      const readButton = document.getElementById('readButton');
+      if (readButton) {
+        readButton.click();
+      }
+    }
+  };
+
   public componentDidMount() {
     this.props.setAsContractInteraction();
     this.props.resetTransactionRequested();
+    setTimeout(
```

Comment:
It would be a good idea to make sure that this timeout is cleared in `componentWillUnmount`

Reply:
Did it. Thanks